File: backend/src/main/java/org/jdk/project/controller/StatisticsController.java
package org.jdk.project.controller;

import lombok.RequiredArgsConstructor;
import org.jdk.project.dto.ListResponse;
import org.jdk.project.dto.statistics.GuideLogDto;
import org.jdk.project.dto.statistics.InquiryStatDto;
import org.jooq.generated.project.tables.pojos.Passenger;
import org.jooq.generated.project.tables.pojos.PassengerAccessTemp;
import org.jdk.project.service.StatisticsQueryService;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/statistics")
@RequiredArgsConstructor
public class StatisticsController {

  private final StatisticsQueryService statisticsQueryService;

  @GetMapping("/in-lounge")
  public ListResponse<Passenger> listInLounge() {
    return statisticsQueryService.listInLoungePassengers();
  }

  @GetMapping("/outgoing")
  public ListResponse<Passenger> listOutgoing() {
    return statisticsQueryService.listOutgoingPassengers();
  }

  @GetMapping("/access-temp")
  public ListResponse<PassengerAccessTemp> listAccessTemp() {
    return statisticsQueryService.listAccessTemps();
  }

  @GetMapping("/inquiry")
  public ListResponse<InquiryStatDto> listInquiry() {
    return statisticsQueryService.listInquiryStats();
  }

  @GetMapping("/guide")
  public ListResponse<GuideLogDto> listGuide() {
    return statisticsQueryService.listGuideLogs();
  }
}

File: backend/src/main/kotlin/org/jdk/project/controller/StatisticsController.kt
package org.jdk.project.controller

import org.jdk.project.dto.ListResponse
import org.jdk.project.dto.statistics.GuideLogDto
import org.jdk.project.dto.statistics.GuideStatisticsQuery
import org.jdk.project.dto.statistics.InquiryStatDto
import org.jdk.project.dto.statistics.InquiryStatisticsQuery
import org.jdk.project.dto.statistics.PassengerRecordDto
import org.jdk.project.dto.statistics.PassengerStatisticsQuery
import org.jdk.project.service.StatisticsQueryService
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.ModelAttribute
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/statistics")
class StatisticsController(
    private val statisticsQueryService: StatisticsQueryService,
) {
    @GetMapping("/in-lounge")
    fun listInLounge(@ModelAttribute query: PassengerStatisticsQuery): ListResponse<PassengerRecordDto> =
        statisticsQueryService.listInLoungePassengers(query)

    @GetMapping("/outgoing")
    fun listOutgoing(@ModelAttribute query: PassengerStatisticsQuery): ListResponse<PassengerRecordDto> =
        statisticsQueryService.listOutgoingPassengers(query)

    @GetMapping("/access-temp")
    fun listAccessTemp(@ModelAttribute query: PassengerStatisticsQuery): ListResponse<PassengerRecordDto> =
        statisticsQueryService.listAccessRecords(query)

    @GetMapping("/inquiry")
    fun listInquiry(@ModelAttribute query: InquiryStatisticsQuery): ListResponse<InquiryStatDto> =
        statisticsQueryService.listInquiryStats(query)

    @GetMapping("/guide")
    fun listGuide(@ModelAttribute query: GuideStatisticsQuery): ListResponse<GuideLogDto> =
        statisticsQueryService.listGuideLogs(query)
}
